Topics/Tech_Stacks/swift.md

```diff
@@ -57,13 +58,71 @@ This information can be changed later, so for starters you can leave `Team` to b
 
 Make sure to use `SwiftUI` and `Swift` as your interface and language respectively, then click `Next` to choose where to store your project, and now you're ready to start.
 
+## Swift View
+A view is a part of a swift application’s user interface. Creating a new project automatically creates a view called content view.  
+
+<img width="283" alt="Screenshot 2023-11-23 at 12 32 33 AM" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/97854264/58e8197c-897b-4bff-be4d-77b6daa62ecd">
+
+Views in Swift are defined as structs and must conform to the View protocol. The content and behavior of the view are provided in the body of the view. To see how the view is transformed into the user’s interface and how users can interact with the view, we can either build and run our application on a simulated device or view the Xcode preview. Refer to the Testing Your App - Simulators: Background section to set up the simulator. An Xcode preview is created by the code defined in lines 22 - 24. It generates a view preview without having to build and run the app. When we open up our project, the preview will appear, as shown below, displaying the view.
```

Comment:
This is just a small thing that might not be necessary but in this paragraph, in the sentence where you mention "lines 22-24", I wasn't sure which image you were referring to so I looked at the image below the paragraph instead of the one above. Adding a portion to the line stating "... in lines 22-24 in the image above" might clarify that for the reader.

Reply:
I see where the confusion can arise. Do you think maybe i should add caption to images maybe naming them so I and other can refer to it more clearly?

Reply:
Yes I think that would help in preventing any confusion!

Reply:
Oh good catch! I think that line numbers aren't visible to the reader unless they go into editing mode, so it would be best to avoid referring to them altogether. Naming your images should resolve this issue!

Reply:
I didn't even think about reader's not being able to read line number. You are write I think it would be more clear to refer to just picture captions instead!